static/js/components/ChannelTracker.js

```diff
@@ -0,0 +1,27 @@
+// @flow
+import React from "react"
+import ReactGA from "react-ga"
+import type { Location } from "react-router"
+import type { Channel } from "../flow/discussionTypes"
+
+type ChannelTrackerProps = {
+  channel: Channel,
+  location: Location
+}
+
+export class ChannelTracker extends React.Component<ChannelTrackerProps> {
+  componentDidMount() {
+    const { channel, location } = this.props
+    if (channel.ga_tracking_id) {
+      const trackerName = channel.ga_tracking_id.replace(/-/g, "_")
+      ReactGA.ga("create", channel.ga_tracking_id, "auto", {
```

Comment:
Is there any issue repeatedly creating this or should you check to see if GA already has it first?

Reply:
No method AFAIK for checking if GA already has it, but this doesn't seem to be an issue. [MM does the same thing](https://github.com/mitodl/micromasters/blob/3c6a4c73dc38e9e590e956d40aef2755cc5b1f92/ui/templates/analytics.html#L18) and in manual testing I haven't noticed any side-effects like duplicated calls.  I'd prefer to initialize all trackers at once along with the main default tracker but that would require having all subscribed + public channels available in `/util/withTracker.js`.